src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private int size = 0;
+    private final int maxItems = 10;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[maxItems];
+        values = (V[]) new Object[maxItems];
+    }
+
+    public StorageImpl(K key, V value) {
+        keys[size] = key;
+        values[size] = value;
+    }
+
     @Override
     public void put(K key, V value) {
+
+        boolean interrupt = false;
+
+        for (int i = 0; i < size; i++) {
+            if (keysEqual(keys[i], key)) {
+                this.values[i] = value;
+                interrupt = true;
+            }
+        }
+
+        if (interrupt == false) {
+            this.keys[size] = key;
+            this.values[size] = value;
+            size++;
+        }
```

Comment:
Better to use the return keyword
```suggestion
        for (int i = 0; i < size; i++) {
            if (keysEqual(keys[i], key)) {
                this.values[i] = value;
                return;
            }
        }

        this.keys[size] = key;
        this.values[size] = value;
        size++;
```